onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h

```diff
@@ -167,6 +167,27 @@ struct ConvTransposeAttributes : public ConvAttributes {
   const std::vector<int64_t> output_shape;
 
  private:
+  int64_t ComputeTotalPad(int64_t in_size, int64_t stride, int64_t adj,
+                          int64_t kernel, int64_t dilation, int64_t out_size) const {
+    int64_t total_pad =
+        std::max<int64_t>(0, (in_size - 1) * stride + adj + (kernel - 1) * dilation + 1 - out_size);
+    return total_pad;
+  }
+
+  void DistributePadding(AutoPadType pad_type, const int64_t& total_pad,
+                         int64_t& pad_head, int64_t& pad_tail) const {
+    if (pad_type == AutoPadType::SAME_UPPER) {  // pad more on head when total_pad is odd.
+      pad_head = total_pad - total_pad / 2;
+      pad_tail = total_pad / 2;
+    } else {
+      // for pad_type is NOTSET, SAME_LOWER or VALID
+      // set pad_head as total_pad/2, pad_tail as total_pad-total_pad/2.
+      // That said, we pad more on tail when total_pad is odd.
+      pad_head = total_pad / 2;
```

Comment:
Make sure the logic for NOTSET, SAME_LOWER or VALID are same here.

Reply:
This has always been as is- I guess only "SAME_UPPER" is special in the way it does the distribution. "SAME_LOWER " is the default way to distribute the padding. So, I guess this is okay.